cram_pr2/cram_pr2_pick_place_demo/src/projection-demo.lisp

```diff
@@ -70,8 +76,37 @@
     (:spoon . ((-0.78 1.5 0.86) (0 0 0 1)))
     (:milk . ((-0.75 1.7 0.95) (0 0 0.7071 0.7071)))))
 
-
-(defun spawn-objects-on-sink-counter (&optional (spawning-poses *object-spawning-poses*))
+(defun make-poses-relative (&optional (spawning-poses *object-spawning-poses*))
+  (let* ((object-types '(:breakfast-cereal :cup :bowl :spoon :milk))
+         (map->surface (cl-transforms:pose->transform
+                       (btr:pose
+                        (btr:rigid-body
+                         (btr:object btr::*current-bullet-world* :kitchen) :|KITCHEN.sink_area_surface|))))
+         
+         (surface->objects (mapcar (lambda (type)
+                                     (setf coordinates (car (cdr (assoc type spawning-poses))))
```

Comment:
This is super ugly. You are defining a global variable `coordinates` right here in the lambda, which you only use later in the lambda. If you need a variable, use a `let` block. If you need a global variable, use a `defparameter` or `defvar` and make sure you name it with `*` on each side: `*my-global-var*`.